Replace Q_ENUMS with individual Q_ENUM

Edward-LeBlanc opened the pull request:
Added sanitizing header fields (remove \r, \n) preventing malicious injection.

Added replacing naked LFs and CRs with CRLF to be compliant with SMTP standard (RFC 5321). 

In SmtpClient.h, replace Q_ENUMS with individual Q_ENUM so that debugging prints names instead of numerical values, added string() function for enum types.

Tested with strict SMTP server.

Edward-LeBlanc committed:
Added sanitizing header fields (remove \r, \n) preventing malicious injection.
Added replacing naked LFs and CRs with CRLF to be compliant with SMTP standard (RFC 5321).
In SmtpClient.h, replace Q_ENUMS with individual Q_ENUM, added string() function for enum types.

Edward-LeBlanc committed:
Corrected choice of variables.
MAde QRegularExpressions static.

Edward-LeBlanc committed:
Made QRegExpressions const, added QRegularExpression::DontCaptureOption.
Made names more expressive.

bluetiger9 commented:
Hi @Edward-LeBlanc! :slightly_smiling_face:

I would say sanitizing user input should be done in the application code, not by the library.

Content wise MIME allows sending basically anything. You just need to change the default `8Bit` content encoding to something that handles special characters / newlines, etc, for ex. `Base64`:
```
part.setEncoding(MimePart::Encoding::Base64)
```

For headers you can use `QuotedPrintable` / `Base64` encoding:
```
message.setHeaderEncoding(MimePart::Encoding::QuotedPrintable); // or ::Base64
```
Thx!

Edward-LeBlanc commented:
I was trying to not repeat code, but I respect your decision. I had already changed the encoding to Base64 in my project, but I apparently missed a couple places that are now fixed. That, of course, is the problem with duplicated code in legacy projects.

What do you think about the part in SmtpClient.h: "replace Q_ENUMS with individual Q_ENUM, added string() function for enum types”? The result of this is seeing the enum names instead of the numeric values in debug and log messages. If you are OK with that, I would revert the other changes.

bluetiger9 commented:
The enum change can remain, it makes perfect sense. Thx!

Edward-LeBlanc committed:
Roll back "sanitization" changes.

Edward-LeBlanc committed:
Fix empty line mismatch in rollback.
